documentation/adr/004-multi-LZA-configuration-management.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: page.njk
+status: Proposed
+date: git Created
+tags:
+  - adr
+order: 4
+title: 004 - Enable Multi LZA Configuration 
+---
+
+**Date: {{ page.date.toISOString().substring(0,10) }}**
+
+**Status: {{ status }}**
+
+## Context
+
+As a result of needing multiple LZA deployments as part of the SDLC of the platform, the products and the ability to assure them, we need a way to be able to manage the environment specific configuration e.g. Workload Accounts.
+
+The files used within LZA to manage the configuration of the deployment are YAML format, with the exception of IAM policies which are JSON format but aren't currently likely to differ between deployments.
+
+Having discussed this requirement with AWS, their recommendation is to use templating as part of the deployment pipeline to pass in environment specific config.
+
+## Decision
+
+We will implement templating before code is deployed to CodeCommit.
+
+We will move to a baseline configuration that is not environment specific and all environments will require their environment specific code to be deployed on top of this configuration.
```

Comment:
We will to having a baseline configuration, with environment specific code being applied on top.

Reply:
Updated